### PR TITLE
TCK reorg & cleanup

### PR DESCRIPTION
**Pull request by vietj**

### CODE REVIEW

File: src/tck/java/io/vertx/codegen/testmodel/CollectionTCK.java [hunk 1]
```diff
@@ -65,6 +65,7 @@ void methodWithHandlerListAndSet(Handler<List<String>> listStringHandler, Handle
   void methodWithHandlerAsyncResultSetDataObject(Handler<AsyncResult<Set<TestDataObject>>> setHandler);
   void methodWithHandlerAsyncResultSetEnum(Handler<AsyncResult<Set<TestEnum>>> setHandler);
 
+  Map<String, String> methodWithMapReturn(Handler<String> handler);
```
Comment (tsegismont) [inline src/tck/java/io/vertx/codegen/testmodel/CollectionTCK.java]: Why is this one needed after all?

File: src/tck/java/io/vertx/codegen/testmodel/TestDataObject.java [hunk 2]
```diff
@@ -55,6 +57,15 @@ public TestDataObject setWibble(double wibble) {
     return this;
   }
 
+  @Override
```
Comment (tsegismont) [inline src/tck/java/io/vertx/codegen/testmodel/TestDataObject.java]: Isn't hashCode needed as well?

File: src/tck/java/io/vertx/codegen/testmodel/CollectionTCK.java (hunk 1 above)
Comment (vietj) [inline src/tck/java/io/vertx/codegen/testmodel/CollectionTCK.java]: it's needed for a test that uses it for doing puts, etc...

File: src/tck/java/io/vertx/codegen/testmodel/TestDataObject.java (hunk 2 above)
Comment (vietj) [inline src/tck/java/io/vertx/codegen/testmodel/TestDataObject.java]: it's not really for putting in a map, it's for testing collection equality

**Pull request merged**